### PR TITLE
Reloads the envelope list when user clicks on the folder they are currently in

### PR DESCRIPTION
**Pull request by StCyr**
Fixes https://github.com/nextcloud/mail/issues/2794

Signed-off-by: Cyrille Bollu <cyr.debian@bollu.be>

### CODE REVIEW

**Comment by GretaD** [comment]
2 things i noticed while testing
1. The "No message" it is shown for 1 second before the message is loaded. 
2. The first message disappears for 1 second and then shown again

ezgif com-video-to-gif(3)

**Comment by StCyr** [comment]
> 2 things i noticed while testing
> 
>     1. The "No message" it is shown for 1 second before the message is loaded.
> 
>     2. The first message disappears for 1 second and then shown again

The first issue is related to https://github.com/nextcloud/mail/issues/2956 and I wonder if we could not simply get rid of the NoMessageSelected  component

The second one, I cannot reproduce

**Comment by GretaD** [comment]
> 
> The first issue is related to #2956 and I wonder if we could not simply get rid of the NoMessageSelected component
> 
oh, then its not because of your PR.

> The second one, I cannot reproduce

oh, interesting, this happens only with drafts, but not because of your PR.

File: src/components/Mailbox.vue [hunk 1]
```diff
@@ -137,6 +137,15 @@ export default {
 		},
 	},
 	watch: {
+		// Reloads the envelope list when user clicks on the folder it's currently in
+		$route(to) {
+			if (to.params.threadId === undefined) {
+				this.loadEnvelopes()
```
[the remark this answers is not in the document]
Comment (StCyr) [reply src/components/Mailbox.vue:143]: Yes, you're right. I'll fix that

File: src/components/Mailbox.vue [hunk 2]
```diff
@@ -137,6 +137,35 @@ export default {
 		},
 	},
 	watch: {
+		// Force the envelope list to re-render when user clicks on the folder it's currently in
+		$route(to) {
+			if (to.params.threadId === undefined) {
+				const first = this.envelopes[0]
+				if (typeof first !== 'undefined') {
+					logger.debug('refreshing mailbox')
+					if (first.flags.draft) {
```
Comment (ChristophWurst) [inline src/components/Mailbox.vue:146]: I have mixed feelings about this. can we take the account's `draftsMailboxId` for a more reliable decision?

Comment (StCyr) [reply src/components/Mailbox.vue:146]: yes, but in this case we should probably also change the test at line 210 (in master):

https://github.com/nextcloud/mail/blob/f3c5de02b387081270bdbc6103c2fe4472e7dc12/src/components/Mailbox.vue#L210

**Comment by miaulalala** [comment]
I just tested this but it doesn't seem to work. I still get the "No Message Selected" component shown when clicking on the inbox the second time. I had a look at the click event on the AppNavigationItem, and that doesn't trigger anything.

**Comment by StCyr** [comment]
> I just tested this but it doesn't seem to work. I still get the "No Message Selected" component shown when clicking on the inbox the second time. I had a look at the click event on the AppNavigationItem, and that doesn't trigger anything.

Strange, it works here.

**Comment by miaulalala** [comment]
Works fine for me now, looks very smooth @StCyr!

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**